app/src/org/commcare/network/CommcareRequestGenerator.java

```diff
@@ -226,17 +233,19 @@ private static String getDigest() {
 
     @Override
     public Response<ResponseBody> simpleGet(String uri) throws IOException {
-        return simpleGet(uri, new HashMap());
+        return simpleGet(uri, new HashMap<>(), new HashMap<>());
     }
 
     @Override
-    public Response<ResponseBody> simpleGet(String uri, Map<String, String> httpParams) throws IOException {
+    public Response<ResponseBody> simpleGet(String uri, Map<String, String> httpParams, Map<String, String> httpHeaders) throws IOException {
+        HashMap<String, String> headers = getHeaders("");
+        headers.putAll(httpHeaders);

```

Comment:
It's best practice to treat these map objects as immutable to ensure that params don't leak between requests [as is done here](https://github.com/dimagi/commcare-android/pull/2322/files#diff-20bb24bd598ae6ae4f34b3bdb1cfbf2aR207)